Userland/Libraries/LibCrypto/Checksum/Adler32.cpp

```diff
@@ -12,10 +12,29 @@ namespace Crypto::Checksum {
 
 void Adler32::update(ReadonlyBytes data)
 {
-    for (size_t i = 0; i < data.size(); i++) {
-        m_state_a = (m_state_a + data.at(i)) % 65521;
-        m_state_b = (m_state_b + m_state_a) % 65521;
+    u64 state_a = m_state_a;
+    u64 state_b = m_state_b;
+    while (data.size()) {
+        // You can verify that no overflow will happen here for at least 10 ** 7 iterations using
```

Comment:
This tracks, but why `1e7`?
The actual upper bound for this is 380368439, which can be obtained by solving the recurrence relation (see mafs).

...so where does 1e7 come from?

---

<details><summary>mafs</summary>
$f(0) = 65520, f(n) = f(n - 1) + 255n + 65520$
$f(n) = {15\over 2} (1+n) (8736+17n)$
$f(n) < 2^{64} \Rightarrow n < {1\over510}\sqrt{37631357910384590012865} - {8753\over34}$

$n < 3.80368439583459e8$
</details>

Reply:
Well, `1e7` was large enough to get rid of this function in the profile. It came from a very rough estimate ($f(n) = 10^5 \cdot n^2$ and $f(n) \le 10^{19} \approx 2^{64}$) of the function you explicitly calculated.

Reply:
Huh, turns out I'm really bad at estimating then.
:shrug: I wasn't asking for a change, was just curious where it came from.